container/openejb-core/src/test/java/org/apache/openejb/resource/AutoConnectionTrackerTest.java

```diff
@@ -214,30 +203,7 @@ public void run() {
 
     // this is a very quick and dirty hack for debugging purpose
     private void assertLogs(final LogCaptureHandler logCapture, final int times, final String message) {
-        final int iteration = 5;
```

Comment:
This might take some time to return, hence the wait... And I suspect the GC is related to that.

Reply:
Actually that wait is not needed here. All logs are captured by LogCaptureHandler. All logs are added synchronously to the handler while calls to business code are executed. 

Asynchronous test code actually waits for all business calls to return so when assertLogs is called to check for specific messages all messages are logged there and there is no need to wait. 

For synchronous calls again there is no need to wait. All code is executed in the main thread.